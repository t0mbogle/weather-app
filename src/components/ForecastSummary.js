import PropTypes from "prop-types";
// import "../styles/App.css";
import React from "react";
import WeatherIcon from "react-icons-weather";

function ForecastSummary(props) {
  const { date, description, icon, temperature } = props;
  const formattedDate = new Date(date).toDateString();
  return (
    // content between the div tags is referencing the forcast.json file.
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary_date">{formattedDate}</div>
      <div className="forecast-summary_icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary_temperature">{temperature.max}ËšC</div>
      <div className="forecast-summary_description">{description}</div>
    </div>
  );
}

export default ForecastSummary;

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};
