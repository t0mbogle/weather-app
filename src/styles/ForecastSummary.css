.forecast-summary {
    border: solid 1px rgb(33, 209, 159);
    background-color: #dfdfdf;
    padding: 8px;
    text-align: center;
}

i {
    font-size: 50px;
    font-weight: 100;
    padding: 20px 10px 20px 10px;
}

.forecast-summary > button {
    font-family: 'Cutive Mono', monospace;
    border: solid 1px rgb(33, 209, 159);
    background-color: #dfdfdf;
    color: black;
    border-radius: 10px;
    padding: 2px 10px 2px 10px;
    margin: 6px 0px 4px 0px;
}

.forecast-summary > button:hover {
    border: solid 1px rgb(2, 120, 120);
    background-color: #eceef1;
}