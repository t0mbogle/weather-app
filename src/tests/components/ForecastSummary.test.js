import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    // asFagment returns a fragment of a rendered component,
    // stub props then get passed in
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );
    // toMatchSnapshot ensures that a value matches the most recent snapshot.
    // Takes a snapshot of the UI and compares it to the test.
    expect(asFragment()).toMatchSnapshot();
  });
});
