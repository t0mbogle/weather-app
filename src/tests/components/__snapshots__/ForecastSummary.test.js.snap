// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      1111111
    </div>
    <div
      class="forecast-summary_description"
    >
      stubIcon
    </div>
    <div
      class="forecast-summary_icon"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary_temperature"
    >
      Stub description
    </div>
  </div>
</DocumentFragment>
`;
